controllers/past-grants/index.js

```diff
@@ -55,72 +83,92 @@ router.use(sMaxAge('1d'), injectBreadcrumbs, (req, res, next) => {
     next();
 });
 
-router.get('/', async (req, res) => {
-    /**
-     * Pick out an allowed list of query parameters for forward on to the grants API
-     * @type {object}
-     */
-    const facetParams = pick(req.query, ['q', 'amount', 'postcode', 'programme', 'year', 'orgType', 'sort']);
-    const queryWithPage = Object.assign({}, facetParams, { page: req.query.page || 1 });
-
-    const data = await request({
-        url: PAST_GRANTS_API_URI,
-        json: true,
-        qs: queryWithPage
-    });
-
-    const commonSortOptions = [
-        {
-            label: 'Oldest first',
-            value: 'awardDate|desc'
-        },
-        {
-            label: 'Lowest amount first',
-            value: 'amountAwarded|asc'
-        },
-        {
-            label: 'Highest amount first',
-            value: 'amountAwarded|desc'
+router
+    .route('/')
+    .post(async (req, res) => {
+        // @TODO how can this handle page numbers / sort options?
+        // Do we need hidden input fields?
+        const queryWithPage = addPaginationParameters(buildAllowedParams(req.body), req.body.page);
+        const grantData = await queryGrantsApi(queryWithPage);
+
+        // Repopulate existing app globals so Nunjucks can read them
+        // outside of Express's view engine context
+        const context = Object.assign({}, res.locals, { grants: grantData.results });
+        const template = path.resolve(__dirname, './views/ajax-results.njk');
+
+        nunjucks.render(template, context, (renderErr, html) => {
+            if (renderErr) {
+                Raven.captureException(renderErr);
+                res.send({
```

Comment:
Thinking about it you can use `res.json` to make sure you have an explicit content-type for this too.